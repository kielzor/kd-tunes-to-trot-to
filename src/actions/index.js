export const addUserFile = audioClip => {
  return {
    type: 'ADD_NEW_FILE',
    audioClip
  };
};

export const storeJog = audioClip => {
  return {
    type: 'ADD_JOG_TRACKS',
    audioClip
  };
};

export const storeWarmup = audioClip => {
  return {
    type: 'ADD_WARMUP_TRACKS',
    audioClip
  };
};

export const storeSprint = audioClip => {
  return {
    type: 'ADD_SPRINT_TRACKS',
    audioClip
  };
};

export const storeWorkout = workout => {
  return {
    type: 'CREATE_WORKOUT',
    workout
  };
};

export const populateWorkout = workout => {
  return {
    type: 'POPULATE_WORKOUT',
    workout
  };
};