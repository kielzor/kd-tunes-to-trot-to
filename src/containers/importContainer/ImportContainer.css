.import-container {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  background-image: url("../../images/Screen Shot 2018-09-20 at 9.50.27 AM.png");
  background-size: cover;
  height: 80vh; }

.import-audio {
  width: 40%;
  height: 20vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center; }

.import-input {
  width: 60%; }

.import-file-input {
  width: 5rem;
  cursor: pointer; }

.import-save-button {
  width: 5rem; }
