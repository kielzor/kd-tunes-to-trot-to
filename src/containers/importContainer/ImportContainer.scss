.import-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 10rem;
}

.import-audio {
  width: 40%;
  height: 20vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.import-input {
  width: 100%;
}

.import-file-input {
  width: 5rem;
  cursor: pointer;
}

.import-save-button {
  width: 5rem;
}
