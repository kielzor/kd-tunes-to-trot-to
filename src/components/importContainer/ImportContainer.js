import React, { Component } from 'react'
import './ImportContainer.css'
import { connect } from 'react-redux'
import { addUserFile } from '../../actions/index'

class ImportContainer extends Component {
	constructor() {
		super()
		this.state = {
			name: '',
			speed: '',
			file: {}
		}
	}

	handleChange = (e) => {
		const { name, value } = e.target

		this.setState ({
			[name]: value
		})
	}

	handleSubmit = (e) => {
		e.preventDefault()
		console.log(this.state)
		this.props.addUserAudioFile(this.state)
	}

	render () {
		return (
			<div className='import-container'>
				<h1>Import Audio</h1>
				<form className='import-audio' onSubmit={this.handleSubmit}>
					<input 
						placeholder='Name' 
						className='import-input'
						name='name'
						value={this.state.value}
						onChange={this.handleChange}
					/>
					<select 
						className='import-input'
						onChange={this.handleChange}
						name='speed'
						value={this.state.value}
					>
						<option default>Speed</option>
						<option>Warmup/ Cooldown</option>
						<option>Jog</option>
						<option>Sprint</option>
					</select>
					<div>
						<input 
							type='file' 
							accept='audio/*' 
							placholder='File' 
							className='import-file-input'
							onChange={this.handleChange}
							name='file'
							value={this.state.value}
						/>
						<button className='import-save-button'>Save</button>
					</div>
				</form>
			</div>
		)
	}
}

const mapDispatchToProps = dispatch => ({
	addUserAudioFile: file => dispatch(addUserFile(file))
})

export default connect(null, mapDispatchToProps)(ImportContainer)