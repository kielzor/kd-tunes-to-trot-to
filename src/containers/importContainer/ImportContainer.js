import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { addUserFile } from '../../actions/index';
import './ImportContainer.css';

export class ImportContainer extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      speed: '',
      file: {}
    };
  }

	handleChange = (event) => {
	  const { name, value } = event.target;

	  this.setState({
	    [name]: value
	  });
	}

	handleSubmit = (event) => {
	  event.preventDefault();
	  this.props.addUserAudioFile(this.state);
	}

	render () {
	  return (
	    <div className='import-container'>
	      <form className='import-audio' onSubmit={this.handleSubmit}>
	      <h1>Import Audio</h1>
	        <input 
	          placeholder='Name' 
	          className='import-input'
	          name='name'
	          value={this.state.value}
	          onChange={this.handleChange}
	        />
	        <select 
	          className='import-input'
	          onChange={this.handleChange}
	          name='speed'
	          value={this.state.value}
	        >
	          <option default>Speed</option>
	          <option>Warmup/ Cooldown</option>
	          <option>Jog</option>
	          <option>Sprint</option>
	        </select>
	        <div>
	          <input 
	            type='file' 
	            accept='audio/*' 
	            placholder='File' 
	            className='import-file-input'
	            onChange={this.handleChange}
	            name='file'
	            value={this.state.value}
	          />
	          <button className='import-save-button'>Save</button>
	        </div>
	      </form>
	    </div>
	  );
	}
}

ImportContainer.propTypes = {
  addUserAudioFile: PropTypes.func
};

export const mapDispatchToProps = dispatch => ({
  addUserAudioFile: file => dispatch(addUserFile(file))
});

export default connect(null, mapDispatchToProps)(ImportContainer);