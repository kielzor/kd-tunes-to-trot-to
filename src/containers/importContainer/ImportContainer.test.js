import React from 'react'
import { ImportContainer, mapDispatchToProps } from './ImportContainer'
import { addUserFile } from '../../actions/index'
import { shallow } from 'enzyme'

describe('ImportContainer', () => {
  let wrapper
  let mockAddUserAudioFile = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<ImportContainer addUserAudioFile={mockAddUserAudioFile}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('handleChange should set the state with the correct values', () => {
    const mockEvent = { target: { name: 'jogLength', value: '1:00' }}

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('jogLength')).toEqual('1:00')
  })

  it('addUserAudioFile should call be called on submit', () => {
    const mockEvent = { preventDefault: jest.fn() }
    
    wrapper.find('.import-audio').simulate('submit', mockEvent)

    expect(mockAddUserAudioFile).toHaveBeenCalled()

  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with addUserFile when addUserAudioFile is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = addUserFile({ name: 'test', speed: 'jog' })

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addUserAudioFile({ name: 'test', speed: 'jog' })

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})