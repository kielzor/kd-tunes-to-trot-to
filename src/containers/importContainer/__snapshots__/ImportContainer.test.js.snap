// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImportContainer
    addUserAudioFile={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Import Audio
</h1>,
        <form
          className="import-audio"
          onSubmit={[Function]}
>
          <input
                    className="import-input"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    value={undefined}
          />
          <select
                    className="import-input"
                    name="speed"
                    onChange={[Function]}
                    value={undefined}
          >
                    <option
                              default={true}
                    >
                              Speed
                    </option>
                    <option>
                              Warmup/ Cooldown
                    </option>
                    <option>
                              Jog
                    </option>
                    <option>
                              Sprint
                    </option>
          </select>
          <div>
                    <input
                              accept="audio/*"
                              className="import-file-input"
                              name="file"
                              onChange={[Function]}
                              placholder="File"
                              type="file"
                              value={undefined}
                    />
                    <button
                              className="import-save-button"
                    >
                              Save
                    </button>
          </div>
</form>,
      ],
      "className": "import-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Import Audio",
        },
        "ref": null,
        "rendered": "Import Audio",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="import-input"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              value={undefined}
/>,
            <select
              className="import-input"
              name="speed"
              onChange={[Function]}
              value={undefined}
>
              <option
                            default={true}
              >
                            Speed
              </option>
              <option>
                            Warmup/ Cooldown
              </option>
              <option>
                            Jog
              </option>
              <option>
                            Sprint
              </option>
</select>,
            <div>
              <input
                            accept="audio/*"
                            className="import-file-input"
                            name="file"
                            onChange={[Function]}
                            placholder="File"
                            type="file"
                            value={undefined}
              />
              <button
                            className="import-save-button"
              >
                            Save
              </button>
</div>,
          ],
          "className": "import-audio",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "import-input",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  default={true}
>
                  Speed
</option>,
                <option>
                  Warmup/ Cooldown
</option>,
                <option>
                  Jog
</option>,
                <option>
                  Sprint
</option>,
              ],
              "className": "import-input",
              "name": "speed",
              "onChange": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Speed",
                  "default": true,
                },
                "ref": null,
                "rendered": "Speed",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Warmup/ Cooldown",
                },
                "ref": null,
                "rendered": "Warmup/ Cooldown",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Jog",
                },
                "ref": null,
                "rendered": "Jog",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sprint",
                },
                "ref": null,
                "rendered": "Sprint",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  accept="audio/*"
                  className="import-file-input"
                  name="file"
                  onChange={[Function]}
                  placholder="File"
                  type="file"
                  value={undefined}
/>,
                <button
                  className="import-save-button"
>
                  Save
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "accept": "audio/*",
                  "className": "import-file-input",
                  "name": "file",
                  "onChange": [Function],
                  "placholder": "File",
                  "type": "file",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Save",
                  "className": "import-save-button",
                },
                "ref": null,
                "rendered": "Save",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Import Audio
</h1>,
          <form
            className="import-audio"
            onSubmit={[Function]}
>
            <input
                        className="import-input"
                        name="name"
                        onChange={[Function]}
                        placeholder="Name"
                        value={undefined}
            />
            <select
                        className="import-input"
                        name="speed"
                        onChange={[Function]}
                        value={undefined}
            >
                        <option
                                    default={true}
                        >
                                    Speed
                        </option>
                        <option>
                                    Warmup/ Cooldown
                        </option>
                        <option>
                                    Jog
                        </option>
                        <option>
                                    Sprint
                        </option>
            </select>
            <div>
                        <input
                                    accept="audio/*"
                                    className="import-file-input"
                                    name="file"
                                    onChange={[Function]}
                                    placholder="File"
                                    type="file"
                                    value={undefined}
                        />
                        <button
                                    className="import-save-button"
                        >
                                    Save
                        </button>
            </div>
</form>,
        ],
        "className": "import-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Import Audio",
          },
          "ref": null,
          "rendered": "Import Audio",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="import-input"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                value={undefined}
/>,
              <select
                className="import-input"
                name="speed"
                onChange={[Function]}
                value={undefined}
>
                <option
                                default={true}
                >
                                Speed
                </option>
                <option>
                                Warmup/ Cooldown
                </option>
                <option>
                                Jog
                </option>
                <option>
                                Sprint
                </option>
</select>,
              <div>
                <input
                                accept="audio/*"
                                className="import-file-input"
                                name="file"
                                onChange={[Function]}
                                placholder="File"
                                type="file"
                                value={undefined}
                />
                <button
                                className="import-save-button"
                >
                                Save
                </button>
</div>,
            ],
            "className": "import-audio",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "import-input",
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    default={true}
>
                    Speed
</option>,
                  <option>
                    Warmup/ Cooldown
</option>,
                  <option>
                    Jog
</option>,
                  <option>
                    Sprint
</option>,
                ],
                "className": "import-input",
                "name": "speed",
                "onChange": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Speed",
                    "default": true,
                  },
                  "ref": null,
                  "rendered": "Speed",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Warmup/ Cooldown",
                  },
                  "ref": null,
                  "rendered": "Warmup/ Cooldown",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Jog",
                  },
                  "ref": null,
                  "rendered": "Jog",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sprint",
                  },
                  "ref": null,
                  "rendered": "Sprint",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    accept="audio/*"
                    className="import-file-input"
                    name="file"
                    onChange={[Function]}
                    placholder="File"
                    type="file"
                    value={undefined}
/>,
                  <button
                    className="import-save-button"
>
                    Save
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "accept": "audio/*",
                    "className": "import-file-input",
                    "name": "file",
                    "onChange": [Function],
                    "placholder": "File",
                    "type": "file",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Save",
                    "className": "import-save-button",
                  },
                  "ref": null,
                  "rendered": "Save",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
